xarray/backends/api.py

```diff
@@ -233,6 +233,101 @@ def maybe_decode_store(store, lock=False):
+        taken from variable attributes (if they exist).  If the `_FillValue` or
+        `missing_value` attribute contains multiple values a warning will be
+        issued and all array values matching one of the multiple values will
+        be replaced by NA.
+    decode_times : bool, optional
+        If True, decode times encoded in the standard NetCDF datetime format
+        into datetime objects. Otherwise, leave them encoded as numbers.
+    concat_characters : bool, optional
+        If True, concatenate along the last dimension of character arrays to
+        form string arrays. Dimensions will only be concatenated over (and
+        removed) if they have no corresponding variable and if they are only
+        used as the last dimension of character arrays.
+    decode_coords : bool, optional
+        If True, decode the 'coordinates' attribute to identify coordinates in
+        the resulting dataset.
+    engine : {'netcdf4', 'scipy', 'pydap', 'h5netcdf', 'pynio'}, optional
+        Engine to use when reading files. If not provided, the default engine
+        is chosen based on available dependencies, with a preference for
+        'netcdf4'.
+    chunks : int or dict, optional
+        If chunks is provided, it used to load the new dataset into dask
+        arrays. This is an experimental feature; see the documentation for more
+        details.
+    lock : False, True or threading.Lock, optional
+        If chunks is provided, this argument is passed on to
+        :py:func:`dask.array.from_array`. By default, a per-variable lock is
+        used when reading data from netCDF files with the netcdf4 and h5netcdf
+        engines to avoid issues with concurrent access when using dask's
+        multithreaded backend.
+    drop_variables: string or iterable, optional
+        A variable or list of variables to exclude from being parsed from the
+        dataset. This may be useful to drop variables with problems or
+        inconsistent values.
+
+    Notes
+    -----
+    This is designed to be fully compatible with `DataArray.to_netcdf`. Saving
+    using `DataArray.to_netcdf` and then loading with this function will
+    produce an identical result.
+
+    All parameters are passed directly to `xarray.open_dataset`. See that
+    documentation for further details.
+
+    See also
+    --------
+    open_dataset
+    """
+    dataset = open_dataset(*args, **kwargs)
+
+    if len(dataset.data_vars) != 1:
+        raise ValueError('Given file dataset contains more than one variable. '
+                         'Please read with xarray.open_dataset and then select '
+                         'the variable you want.')
+    else:
+        data_array, = dataset.data_vars.values()
+
+    data_array._file_obj = dataset._file_obj
+
+    # Reset names if they were changed during saving
+    # to ensure that we can 'roundtrip' perfectly
+    if '__xarray_dataarray_name__' in dataset.attrs:
```

Comment:
Set I would set these names as constants in this module and then import them, e.g.,
```
DATAARRAY_NAME = '__xarray_dataarray_name__'
DATAARRAY_VARIABLE = '__xarray_dataarray_variable__'
```

We have not been so strict about this, but it's usually considered bad form to write the same string literals multiple times in different places.